--- a/qttools/src/designer/src/lib/CMakeLists.txt	2025-09-18 17:43:39.576275700 +0800
+++ b/qttools/src/designer/src/lib/CMakeLists.txt	2025-09-18 20:22:08.700859200 +0800
@@ -180,6 +180,14 @@
         "lib_pch.h"
 )
 
+# Work around QTBUG-95305
+if(CMAKE_GENERATOR STREQUAL "Ninja Multi-Config" AND CMAKE_CROSS_CONFIGS)
+    qt6_wrap_ui(ui_sources_processed ${ui_sources})
+else()
+    set(ui_sources_processed ${ui_sources})
+endif()
+target_sources(Designer PRIVATE ${ui_sources_processed})
+
 # Resources:
 set_source_files_properties("../../../shared/deviceskin/skins/ClamshellPhone.skin"
     PROPERTIES QT_RESOURCE_ALIAS "ClamshellPhone.skin"
